### DISCUSSION TITLE
[celery 5.3] The database backend no longer throws an unlock_chord infinite loop error. ( like mysql )

### DISCUSSION THREAD

**Discussion by paca94** [celery/celery#7691] [General]
**(use translator)**

When using the database backend in version 5.2.7, I saw that an unlock_chord error occurred in an infinite loop and debugged it.

First, when I passed a certain part, I checked to see if the problem occurred.

We have confirmed that the problem occurs the moment app.backend.apply_chord is called in the _chord.run() method.

When the code is executed, it waits for the result for more than 2 tasks, but the tasks are not executed.

So, I checked while printing the stack, and when I made a GroupResult for the result, I confirmed that the results returned by _freeze_group_tasks are evaluated.

Therefore, in the previous code, it was set as regen(x[0] for x in zip(tasks2, results)) , but as the results are evaluated, when the tasks are executed as lazy later, everything except the previously evaluated ones disappears. It was.
(2 are evaluated when if header.tasks is called)

If you actually run the code below in python, you can see that the zip itself is lazy evaluation.
```python3
>>> t = [1,2,3]
>>> t2 = [4,5,6]
>>> zip_r = zip(t, t2)
>>> t2.pop()
6
>>> list(zip_r())
[(1, 4), (2, 5)]
```

After debugging and checking, it was fixed to #7460.  
It was a pity that I could not contribute because it was fixed, but I wrote this article because I wanted others to know that the problem was resolved.

**Comment by auvipy** [comment]
thanks
